app/src/main/java/com/x8bit/bitwarden/ui/platform/components/field/BitwardenPasswordField.kt

```diff
@@ -68,52 +79,82 @@ fun BitwardenPasswordField(
     keyboardType: KeyboardType = KeyboardType.Password,
     imeAction: ImeAction = ImeAction.Default,
     keyboardActions: KeyboardActions = KeyboardActions.Default,
+    textToolbarType: TextToolbarType = TextToolbarType.DEFAULT,
 ) {
     val focusRequester = remember { FocusRequester() }
-    OutlinedTextField(
-        modifier = modifier
-            .tabNavigation()
-            .focusRequester(focusRequester),
-        colors = bitwardenTextFieldColors(),
-        textStyle = BitwardenTheme.typography.sensitiveInfoSmall,
-        label = { Text(text = label) },
-        value = value,
-        onValueChange = onValueChange,
-        visualTransformation = when {
-            !showPassword -> PasswordVisualTransformation()
-            readOnly -> nonLetterColorVisualTransformation()
-            else -> VisualTransformation.None
-        },
-        singleLine = singleLine,
-        readOnly = readOnly,
-        keyboardOptions = KeyboardOptions(
-            keyboardType = keyboardType,
-            imeAction = imeAction,
-        ),
-        keyboardActions = keyboardActions,
-        supportingText = hint?.let {
-            {
-                Text(
-                    text = hint,
-                    style = BitwardenTheme.typography.bodySmall,
+    var textFieldValueState by remember { mutableStateOf(TextFieldValue(text = value)) }
+    val textFieldValue = textFieldValueState.copy(text = value)
+    SideEffect {
+        if (textFieldValue.selection != textFieldValueState.selection ||
+            textFieldValue.composition != textFieldValueState.composition
+        ) {
+            textFieldValueState = textFieldValue
+        }
+    }
+    val textToolbar = when (textToolbarType) {
+        TextToolbarType.DEFAULT -> BitwardenCutCopyTextToolbar(
+            value = textFieldValue,
```

Comment:
We needed the `TextFieldValue` so that we would only copy the selected text instead of all the text. This had a bit of a cascading affect and I ended up copying all of the `textFieldValue `, `SideEffect`, and `lastTextValue` code from `BasicTextField` to ensure the proper functionality.